client/driver/docker.go

```diff
@@ -995,14 +1016,21 @@ func (d *DockerDriver) createContainerConfig(ctx *ExecContext, task *structs.Tas
 			Type:     "volume", // Only type supported
 			ReadOnly: m.ReadOnly,
 		}
-		if m.VolumeOptions != nil {
+		if len(m.VolumeOptions) == 1 {
 			hm.VolumeOptions = &docker.VolumeOptions{
-				NoCopy: m.VolumeOptions.NoCopy,
-				Labels: m.VolumeOptions.Labels,
-				DriverConfig: docker.VolumeDriverConfig{
-					Name:    m.VolumeOptions.DriverConfig.Name,
-					Options: m.VolumeOptions.DriverConfig.Options,
-				},
+				NoCopy: m.VolumeOptions[0].NoCopy,
+			}
+
+			if len(m.VolumeOptions[0].DriverConfig) == 1 {
```

Comment:
m.VolumeOptions[0] could be assigned to a variable as it is used more than once. Same for m.VolumeOptions[0].DriverConfig[0]